File: src/Java/myservicebus-abstractions/src/main/java/com/myservicebus/RequestFaultException.java
package com.myservicebus;

import java.util.stream.Collectors;

public class RequestFaultException extends RuntimeException {
    private final Fault<?> fault;

    public RequestFaultException(String requestType, Fault<?> fault) {
        super("The " + requestType + " request faulted: " +
                fault.getExceptions().stream()
                        .map(m -> {
                            var message = m.getMessage();
                            if (message == null) {
                                message = m.getExceptionType();
                            }
                            return message;
                        })
                        .collect(Collectors.joining(System.lineSeparator())));
        this.fault = fault;
    }

    public Fault<?> getFault() {
        return fault;
    }
}

File: src/MyServiceBus.Abstractions/RequestFaultException.cs
using System;
using System.Linq;

namespace MyServiceBus;

public class RequestFaultException : Exception
{
    public RequestFaultException(string requestType, Fault fault)
        : base($"The {requestType} request faulted: {string.Join(Environment.NewLine, fault.Exceptions.Select(x => x.Message))}")
    {
        RequestType = requestType;
        Fault = fault;
    }

    public string RequestType { get; }

    public Fault Fault { get; }
}
